Content.Server/_Sunrise/EnergyShield/EnergyShieldSystem.cs

```diff
@@ -0,0 +1,64 @@
+using Content.Server.Power.Components;
+using Content.Server.Power.EntitySystems;
+using Content.Server.PowerCell;
+using Content.Server._Sunrise.EnergyShield;
+using Content.Shared.Damage;
+using Content.Shared.Item.ItemToggle.Components;
+using Content.Shared.Item.ItemToggle;
+using Content.Shared.Popups;
+using Content.Shared.Examine;
+using Content.Shared.Timing;
+using Content.Shared.IdentityManagement;
+using Content.Shared.PowerCell.Components;
+using Robust.Shared.Audio.Systems;
+using Robust.Shared.GameStates;
+
+namespace Content.Server._Sunrise.EnergyShield;
+
+public sealed class EnergyShieldSystem : EntitySystem
+{
+    [Dependency] private readonly BatterySystem _battery = default!;
+    [Dependency] private readonly ItemToggleSystem _itemToggle = default!;
+    [Dependency] private readonly SharedPopupSystem _popup = default!;
+    [Dependency] private readonly SharedAudioSystem _audio = default!;
+
+    public override void Initialize()
+    {
+        SubscribeLocalEvent<EnergyShieldComponent, DamageChangedEvent>(OnDamage);
+        SubscribeLocalEvent<EnergyShieldComponent, ItemToggleActivateAttemptEvent>(OnToggleAttempt);
+    }
+
+    private void OnDamage(EntityUid uid, EnergyShieldComponent component, DamageChangedEvent args)
+    {
+        if (!_itemToggle.IsActivated(uid)
+            || args.DamageDelta == null
+            || !TryComp<BatteryComponent>(uid, out var battery))
+            return;
+
+        var cost = args.DamageDelta.GetTotal().Float() * component.EnergyCostPerDamage;
+        if (cost <= 0)
+        return;
```

Comment:
```suggestion
        if (cost <= 0)
        return;
```

Фоматирование поправь. Ретерн уехал